### PR TITLE
Support bytes based on 1024

### PR DESCRIPTION
**Pull request by Goooler**
https://en.wikipedia.org/wiki/Kilobyte

### CODE REVIEW

File: library/src/main/kotlin/me/saket/filesize/FileSize.kt [hunk 1]
```diff
@@ -75,18 +85,21 @@ class FileSize(private val bytes: Long) : Comparable<FileSize> {
...
   override fun toString(): String {
     return when {
-      bytes < 1_000 -> "${bytes.toStringAsFixed()} bytes"
-      bytes < 1_000_000 -> "${(bytes / 1_000.0).toStringAsFixed()} KB"
-      bytes < 1_000_000_000 -> "${(bytes / 1_000_000.0).toStringAsFixed()} MB"
-      else -> "${(bytes / 1_000_000_000.0).toStringAsFixed()} GB"
+      bytes < BytesPerKb -> "${bytes.toStringAsFixed()} bytes"
+      bytes < BytesPerMb -> "${(bytes / 1_000.0).toStringAsFixed()} KB"
+      bytes < BytesPerGb -> "${(bytes / 1_000_000.0).toStringAsFixed()} MB"
+      else -> "${(bytes / BytesPerGb.toDouble()).toStringAsFixed()} GB"
```
Comment (Goooler) [inline library/src/main/kotlin/me/saket/filesize/FileSize.kt:91]: How can we support outputs of `KiB`, `MiB`, and `GiB` here?

**Comment by saket** [comment]
I'm curious, what's your usecase for these?

**Comment by Goooler** [comment]
In some cases, we need to convert units using 1024 as the base, and if we'll land this lib to multiplatform, it's necessary to support this feature on Windows.

**Comment by saket** [comment]
Thanks @Goooler, but in the interest of simplicity, I think I'll skip these APIs unless someone explicitly requests them. In the meantime, extensions can be used if someone needs them.

**Comment by sschuberth** [comment]
> In some cases, we need to convert units using 1024 as the base

Actually, I'd argue that's even in *most* cases. I was rather surprised to see this library only support power-of-10. Usually, using power-of-10 sizes is just the common trick of hard disk vendors to make the capacity of their drives seem larger than they actually are based on power-of-2 sizes.

**Pull request closed without merging**